jyotisha/panchaanga/temporal/festival/applier/solar.py

```diff
@@ -22,6 +25,23 @@ def assign_all(self):
     self.assign_agni_nakshatra()
 
 
+  def assign_sankranti_punyakaala(self):
+    PUNYA_KAALA = {1: (10, 10), 2: (16, 16), 3: (0, 60), 4: (30, 0), 5: (16, 16), 6: (0, 60),
```

Comment:
Users should be able to opt out of having these assigned. So, can add some description_only toml file in adyatithi under a suitable repo - and ensure that this function is active only when some puNyakAla rule is in `self.rules_collection.name_to_rule` . Plz see other methods for example.